### PR TITLE
Implement logr wrapper as part of existing logger and use with clusterctl

### LINKED ISSUE #3105: logs from `clusterctl` library calls are not getting written to the log file as well as kickstart UI
**Bug description**
When creating a management-cluster or a workload cluster, logs from `clusterctl` library calls are not getting written to the log file provided with `--log-file` flag as well as not shown in the kickstart UI.

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**
All clusterctl logs should be shown correctly.

**Steps to reproduce the bug**
- `tanzu management-cluster create --file <> -v 3 --log-file <logfile>`
- Notice the logfile is missing some logs

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Implement logr wrapper as part of existing logger and use with clusterctl
This change will allow `cluster` and `management-cluster` plugin to fetch logs from `clusterctl` library so that those logs can be printed to log file as well as kickstart UI.

### Which issue(s) this PR fixes

Fixes #3105

### Describe testing done for PR
- Verified the log-file output with `tanzu mc create --file <> -v 3 --log-file <log-file-path>` command and `clusterctl` logs are getting printed correctly.
- Sample output:
```
I0808 23:39:56.791181 init.go:115] Validating configuration...
I0808 23:39:56.791776 init.go:181] Using infrastructure provider aws:v1.2.0
I0808 23:39:56.792087 init.go:183] Generating cluster configuration...
I0808 23:39:56.792477 init.go:186] Setting up bootstrapper...
I0808 23:39:56.793513 init.go:196] Bootstrapper created. Kubeconfig: /Users/anujc/.kube-tkg/tmp/config_6Zlsh8Cy
I0808 23:39:56.816175 init.go:215] Installing providers on bootstrapper...
I0808 23:39:56.863403 inventory.go:175] Installing the clusterctl inventory CRD
I0808 23:39:56.954875 init.go:97] Fetching providers
I0808 23:39:57.990234 cert_manager.go:174] Installing cert-manager Version="v1.7.2"
I0808 23:39:59.827764 cert_manager.go:536] Waiting for cert-manager to be available...
^[[1;2C^[[1;2CI0808 23:40:35.242658 installer.go:104] Installing Provider="cluster-api" Version="v1.1.5" TargetNamespace="capi-system"
I0808 23:40:35.243905 installer.go:108] Creating objects Provider="cluster-api" Version="v1.1.5" TargetNamespace="capi-system"
I0808 23:40:36.834581 installer.go:113] Creating inventory entry Provider="cluster-api" Version="v1.1.5" TargetNamespace="capi-system"
I0808 23:40:36.936235 installer.go:104] Installing Provider="bootstrap-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-bootstrap-system"
I0808 23:40:36.938359 installer.go:108] Creating objects Provider="bootstrap-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-bootstrap-system"
I0808 23:40:38.251125 installer.go:113] Creating inventory entry Provider="bootstrap-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-bootstrap-system"
I0808 23:40:38.301780 installer.go:104] Installing Provider="control-plane-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-control-plane-system"
I0808 23:40:38.303577 installer.go:108] Creating objects Provider="control-plane-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-control-plane-system"
I0808 23:40:40.627647 installer.go:113] Creating inventory entry Provider="control-plane-kubeadm" Version="v1.1.5" TargetNamespace="capi-kubeadm-control-plane-system"
I0808 23:40:40.856080 installer.go:104] Installing Provider="infrastructure-aws" Version="v1.2.0" TargetNamespace="capa-system"
I0808 23:40:40.857033 installer.go:108] Creating objects Provider="infrastructure-aws" Version="v1.2.0" TargetNamespace="capa-system"
I0808 23:40:47.056148 installer.go:113] Creating inventory entry Provider="infrastructure-aws" Version="v1.2.0" TargetNamespace="capa-system"
I0808 23:40:47.114856 init.go:541] installed  Component=="cluster-api"  Type=="CoreProvider"  Version=="v1.1.5"
I0808 23:40:47.116720 init.go:541] installed  Component=="kubeadm"  Type=="BootstrapProvider"  Version=="v1.1.5"
I0808 23:40:47.118223 init.go:541] installed  Component=="kubeadm"  Type=="ControlPlaneProvider"  Version=="v1.1.5"
I0808 23:40:47.119803 init.go:541] installed  Component=="aws"  Type=="InfrastructureProvider"  Version=="v1.2.0"
```

### Release note

```release-note
Implement logr wrapper as part of existing logger and use with clusterctl
```

### CODE REVIEW

**Review by marckhouzam — COMMENTED** [review]
Nice.
A couple of questions around backwards-compatibility, but the it looks good from my newbie point-of-view 😄 .  I didn't run any tests.

File: pkg/v1/tkg/log/interface.go [hunk 1]
```diff
@@ -66,8 +66,12 @@ type LoggerImpl interface {
...
 	// The logger will write only log messages with a level/V(x) equal or higher to the threshold.
 	SetThreshold(threshold *int32)
-
+	// Clone creates cloned copy of the logger
+	Clone() LoggerImpl
+	// CloneWithLevel creates cloned copy of the logger with updated log level
 	CloneWithLevel(level int) LoggerImpl
+	// WithCallDepth implements a New Option that allows to set the callDepth level for a logger.
+	WithCallDepth(callDepth int) LoggerImpl
 }
```
[the remark this answers is not in the document]
Comment (anujc25) [reply pkg/v1/tkg/log/interface.go:75]: It's a good question. Currently we are the only consumer of this interface and hence it is safe to update this interface and this additional functionality for this specific usecase. But we need to be careful in future when other modules starts consuming this interface.

File: pkg/v1/tkg/log/logr.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package log
+
+import "github.com/go-logr/logr"
+
+// GetLogr logs a warning messages with the given message format with format specifier and arguments.
+func GetLogr() logr.Logger {
```
[the remark this answers is not in the document]
Comment (anujc25) [reply pkg/v1/tkg/log/logr.go:9]: Same as other, Currently we are the only consumer of this function and hence it is safe to update the return type of this. But we need to be careful in future when other modules starts consuming this.

**Review by vuil — APPROVED** [review]
lgtm, thanks.

**Pull request merged**